### ISSUE TITLE
Defining characters placeholders in addition to word placeholders

### ISSUE THREAD

**Issue by kurtespinosa** [kyle29/Semantic-Textual-Similarity-Tensorflow#1]
Hi Kyle29, thanks for the code. In the computation graph definitions, you have this for the words in sentences:

sentences_A=tf.placeholder(tf.int32,shape=([None,FLAGS.max_length]),name='sentences_A')
sentencesA_length=tf.placeholder(tf.int32,shape=([None]),name='sentencesA_length')

I wonder what would be the definition if you include the character level? Would it be like this?

sentences_words_A=tf.placeholder(tf.int32,shape=([None,FLAGS.max_length, max_word_len,]),name='sentences_A')
sentencesA_words_length=tf.placeholder(tf.int32,shape=([None, FLAGS.max_length]),name='sentencesA_length')

Thanks!

**Comment by kwangh** [comment]
Hi kurtespinosa.
If you are looking for a similarity between word pairs in character level, you need to make a character level embeddings instead of word embeddings. After changing original word embeddings to the character level embeddings, you can use the original code
<
  sentences_A=tf.placeholder(tf.int32,shape=([None,FLAGS.max_length]),name='sentences_A')
  sentencesA_length=tf.placeholder(tf.int32,shape=([None]),name='sentencesA_length')
>
to measure the similarity between word pairs.
I can recommend you this paper
Mueller, Jonas, and Aditya Thyagarajan. "Siamese Recurrent Architectures for Learning Sentence Similarity." AAAI. 2016.
http://anthology.aclweb.org/W16-1617

**Comment by kurtespinosa** [comment]
Thank you very much Kyle29 for your reply. I have a question: why do you have start_token and end_token in your input representation? What is it for? Sorry I haven't gone thru your code thoroughly yet. 

-- Kurt

**Comment by kwangh** [comment]
If I am right, the batch normalization's start and end token(is this the one you are askying for?) are each beginning and end indexes of batch inputs.
`for i in range(total_batch):`
`start=i*config.batch_size`
`end=(i+1)*config.batch_size`
